src/pages/distroSettings/tabs/ProviderTab/schemaFields.ts

```diff
@@ -241,32 +245,201 @@ const mountPoints = {
+  schema: {
+    buildType,
+    imageUrl,
+    registryUsername,
+    registryPassword,
+    mergeUserData,
+    userData,
+    securityGroups,
+  },
+  uiSchema: {
+    buildType: {
+      "ui:allowDeselect": false,
+    },
+    registryUsername: {
+      "ui:optional": true,
+    },
+    registryPassword: {
+      "ui:optional": true,
+      "ui:inputType": "password",
+    },
+    mergeUserData: {
+      "ui:elementWrapperCSS": mergeCheckboxCSS,
+    },
+    userData: {
+      "ui:widget": "textarea",
+      "ui:elementWrapperCSS": userDataCSS,
+      "ui:rows": 6,
+    },
+    securityGroups: {
+      "ui:addButtonText": "Add security group",
+      "ui:orderable": false,
+    },
+  },
 };
 
 export const ec2FleetProviderSettings = {
-  amiId,
-  instanceType,
-  sshKeyName,
-  fleetOptions,
-  instanceProfileARN,
-  mergeUserData,
-  userData,
-  securityGroups,
-  vpcOptions,
-  mountPoints,
+  schema: {
+    amiId,
+    instanceType,
+    sshKeyName,
+    fleetOptions,
+    instanceProfileARN,
+    mergeUserData,
+    userData,
+    securityGroups,
+    vpcOptions,
+    mountPoints,
+  },
+  uiSchema: {
+    amiId: {
+      "ui:placeholder": "e.g. ami-1ecba176",
+    },
+    instanceType: {
+      "ui:description": "EC2 instance type for the AMI. Must be available.",
+      "ui:placeholder": "e.g. t1.micro",
+    },
+    sshKeyName: {
+      "ui:description": "SSH key to add to the host machine.",
+    },
+    fleetOptions: {
+      fleetInstanceType: {
+        "ui:allowDeselect": false,
+      },
+      useCapacityOptimization: {
+        "ui:data-cy": "use-capacity-optimization",
+        "ui:bold": true,
+        "ui:description":
+          "Use the capacity-optimized allocation strategy for spot (default: lowest-cost)",
+        "ui:elementWrapperCSS": capacityCheckboxCSS,
+      },
+    },
+    mergeUserData: {
+      "ui:elementWrapperCSS": mergeCheckboxCSS,
+    },
+    userData: {
+      "ui:widget": "textarea",
+      "ui:elementWrapperCSS": userDataCSS,
+      "ui:rows": 6,
+    },
+    securityGroups: {
+      "ui:addButtonText": "Add security group",
+      "ui:orderable": false,
+    },
+    vpcOptions: {
+      useVpc: {
+        "ui:data-cy": "use-vpc",
+      },
+      subnetId: {
+        "ui:placeholder": "e.g. subnet-xxxx",
+        "ui:elementWrapperCSS": indentCSS,
+      },
+      subnetPrefix: {
+        "ui:description":
+          "Looks for subnets like <prefix>.subnet_1a, <prefix>.subnet_1b, etc.",
+        "ui:elementWrapperCSS": indentCSS,
+      },
+    },
+    mountPoints: {
+      "ui:data-cy": "mount-points",
+      "ui:addButtonText": "Add mount point",
+      "ui:orderable": false,
+      "ui:topAlignDelete": true,
+      items: {
+        "ui:ObjectFieldTemplate": AccordionFieldTemplate,
+        "ui:numberedTitle": "Mount Point",
+      },
+    },
+  },
+};
+
+export const ec2OnDemandProviderSettings = {
+  schema: {
+    amiId,
+    instanceType,
+    sshKeyName,
+    instanceProfileARN,
```

Comment:
[nit] The legacy platform mentions that ARN = Amazon Resource Name, so maybe that would be worth including in the description